src/components/SettingsAdvanced.vue

```diff
@@ -0,0 +1,230 @@
+        };
+    },
+    computed: {
+        filteredSettings() {
+            let settings = this.settings;
+            let out = [];
+            Object.values(settings).forEach((value, index) => {
+                if (value.key.toLowerCase().includes(this.filterString.toLowerCase())) {
+                    out.push(value);
+                }
+            });
+            return out;
+        },
+        settings() {
+            let out = {};
+            let base = [];
+            this.buildTree(out, base, state.getSetting('settings'), false);
+            this.buildTree(out, base, state.getSetting('user_settings'), true);
+
+            return _.orderBy(Object.values(out), [
+                o => o.key.split('.').length - 1,
+                'key',
+            ], ['asc']);
+        },
+    },
+    methods: {
+        resetValue(event, settingKey) {
+            let newVal = state.getSetting('settings.' + settingKey);
+            if (!newVal) {
+                newVal = null;
+            }
+            state.setting(settingKey, newVal);
+        },
+        blurOnEnter(event) {
+            if (event.keyCode === 13) {
+                event.target.blur();
+            }
+        },
+        updateSetting(event, settingKey) {
+            let target = event.target;
+            let val = target.type === 'checkbox' ? target.checked : target.value;
+            switch (target.type) {
+            case 'checkbox':
+                val = target.checked;
+                break;
+            case 'number':
+                val = parseInt(target.value, 10);
+                break;
+            default:
+                val = target.value;
+                break;
+            }
+            if (state.setting(settingKey) === val) {
+                return;
+            }
+
+            state.setting(settingKey, val);
+        },
+        buildTree(data, base, object, modified) {
+            Object.entries(object).forEach(([key, value]) => {
+                let ourBase = base.concat([key]);
+                if (['string', 'boolean', 'number'].includes(typeof value)) {
+                    if (this.ignoreKeys.includes(key) ||
+                     (ourBase[0] && this.ignoreKeys.includes(ourBase[0]))) {
+                        return;
+                    }
+
+                    if (!data[ourBase.join('.')] || data[ourBase.join('.')].val !== value) {
+                        data[ourBase.join('.')] = {
+                            key: ourBase.join('.'),
+                            val: value,
+                            type: typeof value,
+                            modified: modified,
+                        };
+                    }
+                } else if (typeof value === 'object') {
+                    this.buildTree(data, ourBase, value, modified);
+                }
+            });
+        },
+    },
+};
+</script>
+
+<style>
+
+.kiwi-appsettings-block-advanced {
```

Comment:
If this is styling something in the appsettings component, then this style should be in that component rather than here.